cli/internal/runcache/runcache.go

```diff
@@ -16,48 +17,35 @@ import (
 	"github.com/vercel/turborepo/cli/internal/globby"
 	"github.com/vercel/turborepo/cli/internal/nodes"
 	"github.com/vercel/turborepo/cli/internal/ui"
+	"github.com/vercel/turborepo/cli/internal/util"
 )
 
-// LogsMode defines the ways turbo can output logs during a run for cached and not cached tasks
-type LogsMode int
-
 // LogReplayer is a function that is responsible for replaying the contents of a given log file
 type LogReplayer = func(logger hclog.Logger, output cli.Ui, logFile fs.AbsolutePath)
 
-const (
-	// FullLogs - show all,
-	FullLogs LogsMode = iota
-	// HashLogs - only show task hash
-	HashLogs
-	// NoLogs - show nothing
-	NoLogs
-)
-
-const (
-	logsModeFull     = "full"
-	logsModeHashOnly = "hash-only"
-	logsModeNewOnly  = "new-only"
-	logsModeNone     = "none"
-)
-
 // Opts holds the configurable options for a RunCache instance
 type Opts struct {
-	SkipReads         bool
-	SkipWrites        bool
-	CacheHitLogsMode  LogsMode
-	CacheMissLogsMode LogsMode
-	LogReplayer       LogReplayer
+	SkipReads              bool
+	SkipWrites             bool
+	TaskOutputModeOverride string
```

Comment:
I used a string here to abuse the zero-value of the string to indicate whether or not the value was overridden. The value is only set when the flag is present, so, it's an empty string otherwise. I also considered using a `bool` to mark the override and the actual type but that seemed excessive.